src/main/java/de/tum/cit/aet/artemis/programming/service/ProgrammingExerciseBuildConfigService.java

```diff
@@ -0,0 +1,161 @@
+package de.tum.cit.aet.artemis.programming.service;
+
+import static de.tum.cit.aet.artemis.core.config.Constants.MAX_ENVIRONMENT_VARIABLES_DOCKER_FLAG_LENGTH;
+import static de.tum.cit.aet.artemis.core.config.Constants.PROFILE_CORE;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import jakarta.annotation.Nullable;
+
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Service;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import de.tum.cit.aet.artemis.buildagent.dto.DockerRunConfig;
+import de.tum.cit.aet.artemis.programming.domain.ProgrammingExerciseBuildConfig;
+
+@Profile(PROFILE_CORE)
+@Service
+public class ProgrammingExerciseBuildConfigService {
+
+    private static final Logger log = org.slf4j.LoggerFactory.getLogger(ProgrammingExerciseBuildConfigService.class);
+
+    /**
+     * Converts a JSON string representing Docker flags (in the form of a list of key-value pairs)
+     * into a {@link DockerRunConfig} instance.
+     *
+     * <p>
+     * The JSON string is expected to represent a list of key-value pairs where each
+     * entry is a list containing two strings: the first being the key and the second being the value.
+     * Example JSON input:
+     *
+     * <pre>
+     * [["network", "none"], ["env", "TEST"]]
```

Comment:
General question regarding the design of the JSON structure (blocker, since chaning the structure later can only be done with a Java-based migration rather than a pure SQL migration, so getting the structure right now is important):
Would it make sense to use a more structured JSON that can more easily be extended in the future with additional values?
Then for example the ENV variables would only have to be parsed once initially from the user input into a DTO. This DTO can then be directly serialised and deserialised to/from the JSON stored in the database without having to do the complex (and error-prone) regex-based parsing.

For example:
```json
{
   "network": true,
   "env": {
     "SOME_ENV_VAR": "some-value-for-env-var",
     "ANOTHER_ENV_VAR": "some-other-value"
   }
}
```

This would then also enable something like this https://www.baeldung.com/hibernate-persist-json-object where the (de)serialisation is managed by Hibernate automatically using an `AttributeConverter`.

Reply:
I changed the structure to JSON and removed the regex. However, I didnt add the attributeConverter since we don't want the parsing to happen everytime we fetch from the database (since we actually rarely need the dockerFlags)